### PR TITLE
Fix executor setting its own hostname

### PR DESCRIPTION
**Pull request by croissanne**
Cloud-init data was broken + the script wasn't executable :facepalm:

### CODE REVIEW

**Review by ezr-ondrej — DISMISSED** [review]
I don't understand fully, but the change looks quite sane :)

**Review by achilleas-k — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -481,7 +474,7 @@ filesystem.sh:
     - RUNNER:
       # don't execute on fedora-39 yet until https://lkml.kernel.org/linux-xfs/20230823220225.3591135-1-david@fromorbit.com/#t is in the kernel
```
Comment (elkoniu) [inline .gitlab-ci.yml:475]: I guess this comment can be removed now :)

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -573,8 +565,7 @@ API-module-hotfixes:
   parallel:
     matrix:
       - RUNNER:
-          - rhos-01/fedora-38-x86_64
-          # - rhos-01/fedora-39-x86_64
```
Comment (elkoniu) [inline .gitlab-ci.yml:577]: Maybe we should add comment here why 39 is skipped (if this is the same case as in line 483?)

**Review by elkoniu — COMMENTED** [review]
Generally LGTM, two minor comments.

**Comment by ezr-ondrej** [comment]
Failing on Fedora 40 if I'm reading it correctly.

File: .gitlab-ci.yml (hunk 1 above)
Comment (croissanne) [reply .gitlab-ci.yml:475]: removed, and switched to 40 since it seemed to fail still

File: .gitlab-ci.yml (hunk 2 above)
Comment (croissanne) [reply .gitlab-ci.yml:577]: there's no openstack image for fedora39 currently

**Review by elkoniu — COMMENTED** [review]
LGTM, but as an Golang moron I will leave approve to some Golang native speakers :)

**Comment by jrusz** [comment]
There is something wrong with the filesystem.sh test on fedora-40, it seem image-info has issue there, I'd wait with the transition for now to fix the stage runners.

**Comment by ezr-ondrej** [comment]
Yeah I'd agree, otherwise we'd need to get it in broken to fix stage, right?

**Comment by jrusz** [comment]
> Yeah I'd agree, otherwise we'd need to get it in broken to fix stage, right?

Well breaking `main` is not desirable so I'd say otherwise we'd need to debug and fix the issue which would delay the fix for stage :)

**Comment by jrusz** [comment]
> > There is something wrong with the filesystem.sh test on fedora-40, it seem image-info has issue there, I'd wait with the transition for now to fix the stage runners.
> 
> Can't run fedora-38 anymore since the rpmrepo snapshots got deleted, so main currently is broken already.

Yeah well in that case I'd just not run it for now and fix in a followup... I read the comments and it seems there was something missing in the kernel, we should check if it's there already.

**Comment by croissanne** [comment]
> > > There is something wrong with the filesystem.sh test on fedora-40, it seem image-info has issue there, I'd wait with the transition for now to fix the stage runners.
> > 
> > 
> > Can't run fedora-38 anymore since the rpmrepo snapshots got deleted, so main currently is broken already.
> 
> Yeah well in that case I'd just not run it for now and fix in a followup... I read the comments and it seems there was something missing in the kernel, we should check if it's there already.

the issue now seems to be image-info:
```
Traceback (most recent call last):
  File "/usr/libexec/osbuild-composer-test/image-info", line 2891, in <module>
    main()
  File "/usr/libexec/osbuild-composer-test/image-info", line 2885, in main
    report = analyse_image(target)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/libexec/osbuild-composer-test/image-info", line 2731, in analyse_image
    append_partitions(report, target)
  File "/usr/libexec/osbuild-composer-test/image-info", line 2706, in append_partitions
    append_filesystem(report, root_tree)
  File "/usr/libexec/osbuild-composer-test/image-info", line 2317, in append_filesystem
    dnf_conf = read_dnf_conf(tree)
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/libexec/osbuild-composer-test/image-info", line 2158, in read_dnf_conf
    dnf_config = _read_inifile_to_dict(f"{tree}/etc/dnf/dnf.conf")
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/libexec/osbuild-composer-test/image-info", line 2124, in _read_inifile_to_dict
    parser.readfp(f)
    ^^^^^^^^^^^^^
AttributeError: 'RawConfigParser' object has no attribute 'readfp'. Did you mean: 'read'?
ERROR image-info failed, show last few kernel message to debug
```

**Review by thozza — APPROVED** [review]
LGTM

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -479,9 +474,7 @@ filesystem.sh:
   parallel:
     matrix:
     - RUNNER:
-      # don't execute on fedora-39 yet until https://lkml.kernel.org/linux-xfs/20230823220225.3591135-1-david@fromorbit.com/#t is in the kernel
-      # see also https://github.com/osbuild/osbuild-composer/pull/3820#issuecomment-1881587335 for more details
-      - aws/fedora-38-x86_64
+    # TODO add fedora runners here
```
Comment (ezr-ondrej) [inline .gitlab-ci.yml:477]: Check runners script is not happy about this I'm gussing:
`AttributeError: 'NoneType' object has no attribute 'split'`

Comment (croissanne) [reply .gitlab-ci.yml:477]: :cry:

**Comment by ezr-ondrej** [comment]
Are we ok with being out of date, can someone force rebase merge if the tests pass?

**Comment by croissanne** [comment]
> Are we ok with being out of date, can someone force rebase merge if the tests pass?

composer doesn't require prs to be rebased becuse of CI taking a long time. But yes I can always force merge too

**Comment by croissanne** [comment]
filesystem.sh succeeded + WorkerExecutor did, merging to get the image build rolling, will follow up with an image-info fix.

**Pull request merged**